queue/worker-interaction/index.md

```diff
@@ -0,0 +1,60 @@
+---
+layout:             default
+class:              markdown
+sequence_diagrams:  true
+---
+
+Queue - Worker Interaction (draft)
+==================================
+This document outlines a how queue and worker can interact, and how Azure Queue
+Storage queues participates in this interaction. We refer to TaskCluster Queue
+simply as _queue_, where as Azure Queue Storage queues, will be referred to as
+_Azure queues_.
+
+
+Polling for Tasks
+=================
+The simplest way to poll for tasks is to call `claimWork` on the queue.
+The queue will then poll the underlying Azure queue, and claim a task for you
+if one is available. However, this won't well if a lot of workers are
+constantly polling the queue.
+
+**Proposal,** the queue could track number of `claimWork` requests per 10s, and
+return exponentially large back-off times to workers. Though this is complicated
+to implement.
+
+The efficient way to poll for tasks is to call `pollTaskUrl`. This returns a
+signed URL for an Azure queue that you can poll for tasks
+(Azure queues can handle 2k req. / s). The signed Azure queue URL will return
+XML, the worker does not need to parse this XML. Just grep the response text
```

Comment:
Do you suggest we do XML parsing on the worker... It would allow us to poll for multiple tasks at once... Though claim with TC queue will like require multiple calls afterwards...